plugin.video.montreal.greek-tv/resources/lib/default.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+
+import urllib.parse, os, sys
+import xbmcvfs, xbmcgui, xbmcplugin, xbmcaddon
+
+addon_id       = 'plugin.video.montreal.greek-tv'
+Home           = xbmcvfs.translatePath(xbmcaddon.Addon().getAddonInfo('path'))
+art            = xbmcvfs.translatePath(Home + '/resources/art/')
+icon           = xbmcvfs.translatePath(os.path.join(Home, 'icon.png'))
+tv_icon        = xbmcvfs.translatePath(os.path.join(art , 'montreal_tv.png'))
+radio_icon     = xbmcvfs.translatePath(os.path.join(art , 'montreal_radio.png'))
+radio_fanart   = xbmcvfs.translatePath(os.path.join(art , 'montreal_backround.jpg'))
+
+__handle__     = int(sys.argv[1])
+
+TV_URL = 'http://live.greektv.ca/hls1/greektv.m3u8'
+RADIO_URL = 'http://live.greekradio.ca:8000/live'
+
+def Index():
+    addDir("Montreal Greek TV", TV_URL, 2, tv_icon, radio_fanart)
+    addDir("Montreal Greek Radio", RADIO_URL, 2, radio_icon, radio_fanart)
+    xbmcplugin.endOfDirectory(__handle__)
+
+def addDir(name, url, mode, iconimage, fanart):
+        u = sys.argv[0]+"?url="+urllib.parse.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.parse.quote_plus(name)+"&iconimage="+urllib.parse.quote_plus(iconimage)+"&fanart="+urllib.parse.quote_plus(fanart)
+        liz = xbmcgui.ListItem(name)
```

Comment:
Please use `offscreen=True` as initializer kwarg, it will speed up your add-on a bit

Reply:
added